### PR TITLE
fix(signin): displaying new message for already verified sign-in tokens

### PR DESCRIPTION
**Pull request by divyabiyani**
Test cases added as was discussed in #4173 .
Also,squashed the commits and put them together as per the guidelines.

### CODE REVIEW

**Comment by divyabiyani** [comment]
@vbudhram Please check this one.

**Comment by vbudhram** [comment]
@divyabiyani Thanks for PR! LGTM 👍

File: app/scripts/models/verification/base.js [hunk 1]
```diff
@@ -99,6 +99,14 @@ define(function (require, exports, module) {
       return !! this._isExpired;
     },
 
+    markUsed: function () {
```
Comment (vladikoff) [inline app/scripts/models/verification/base.js]: missing jsdoc like the rest of the functions in this file.

Comment (divyabiyani) [inline app/scripts/models/verification/base.js]: @vladikoff I will add it.

Comment (vbudhram) [inline app/scripts/models/verification/base.js]: Woops, good catch @vladikoff

**Pull request merged**